### DISCUSSION TITLE
Providing a prior to the default optimization algorithm

### DISCUSSION THREAD

**Discussion by clementruhm** [google/vizier#1205] [Q&A]
Is it possible to provide some historical data to the default optimization algorithm so it converges better? I've seen some low-level API in the codebase, but cant find any tutorial for that using client API. Can you suggest what would be the approach and how much it can help?

Another question - is it possible to provide a more high-level prior knowledge derived from previous runs? for example, is it possible to somehow inject feature importance for the sampling algorithm? For example increase initial variance for the more important features, so Vizier tends to explore those features first. How can I approach such a problem?

Would appreciate any references or links to get up to speed with algo / codebase to understand better how this can be done.

Thanks in advance

**Comment by clementruhm** [comment]
I can see there is a function: https://github.com/google/vizier/blob/1fc8537b9371edf432d0d69fb15ad3d698718daa/vizier/_src/algorithms/designers/gp_bandit.py#L281

is there way to use it with client API? is there example how to use it?

Should I create trials but hand, complete them and then call "set_priors"? Smth like:

```python
trials_lst = [vz.trialSuggestion({"rooms": 1})]
completed_lst = [x.to_trial().complete(vz.Measurement(metrics={"price": 10000}))]
completed = vza.Completedtrials(trials=completed_lst)
 
# how to get access to designer??
client.get_designer().set_priors(completed)
```

**Comment by xingyousong** [comment]
**provide some historical data to the default optimization algorithm so it converges better?**
* Use [`study_client.add_trial()`](https://github.com/google/vizier/blob/main/vizier/_src/service/clients.py#L152). 

**possible to somehow inject feature importance**
* Not really - The Vizier GP computes the scaling of each feature (via ARD), and will compute the importances on its own. If you already know which features are important, it might be better off performing a manual search yourself initially, and then seeding this into Vizier GPBandit.
